Implement `Sub-Zero Smash` mob TP skill.

mrhappyasthma opened the pull request:
The implementation is pretty straightforward since it's a single-hit physical attack.

The potency of the paralysis is taken from: https://www.bg-wiki.com/ffxi/Sub-zero_Smash

Also updated the range and shape on the .sql to match the documentation. It should be a cone shape of 5 yalms.

<!-- remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm: -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

mrhappyasthma committed:
Implement `Sub-Zero Smash` mob TP skill.

The implementation is pretty straightforward since it's a single-hit physical attack.

The potency of the paralysis is taken from: https://www.bg-wiki.com/ffxi/Sub-zero_Smash

Also updated the range and shape on the .sql to match the documentation. It should be a cone shape of 5 yalms.

claywar commented on scripts/globals/mobskills/sub-zero_smash.lua:
```diff
@@ -0,0 +1,38 @@
+
+mobskill_object.onMobSkillCheck = function(target, mob, skill)
+    return 0
+end
```
Caught it before I could comment on it 👍

mrhappyasthma replied:
;)

mrhappyasthma committed:
Inline unessential local vars.

Per the discussion on the other pull request, I inlined the local vars that were not particularly useful: they weren't reused nor helped decrease line lengths in any meaningful way.

mrhappyasthma commented:
Primary part of https://github.com/LandSandBoat/server/issues/201

zach2good reviewed (approved):
Seems reasonable, LGTM 👍

mrhappyasthma committed:
Merge branch 'base' into subzerosmash

mrhappyasthma commented:
The github web UI tools won't seem to let me resolve this, so I'll fix it once I'm home from work and push a commit to resolve the conflict.

mrhappyasthma commented:
Actually it looks like it did work. It was just super delayed to reflect.

TeoTwawki commented:
yeah I'd avoid the webUI as much as possible for all but the simplest of 1 line changes.... Its often the source of much pain for peopel who try to use it.

TeoTwawki commented on scripts/globals/mobskills/sub-zero_smash.lua:
```diff
@@ -0,0 +1,35 @@
+--         still be used anytime. As a result, any other Ruszor spells
+--         should not trigger if the target is behind the mob.
+-----------------------------------
+require("scripts/globals/settings")
```
Doesn't look like this settings require is used.

I tend to do widest line 1st unless I have a logic reason not to so:
```lua
-----------------------------------
require("scripts/globals/monstertpmoves")
require("scripts/globals/status")
-----------------------------------
```
?

mrhappyasthma replied:
I was going by the (now many-times-deprecated) docs from Darkstar: https://wiki.dspt.info/index.php/How_to_Add_TP_Moves

```
Always:

require("/scripts/globals/settings");
require("/scripts/globals/status");
require("/scripts/globals/monstertpmoves");
```

I'm happy to remove unused requires.

I typically favor alphabetizing the imports, but I'm open to list them in an order here. I don't have a strong preference.

TeoTwawki replied:
I've been murdering extra requires every chance I get for a long time now, but old docs have copypasta from old files unfortunately and the current project wiki needs lots of love still.

I don't think we have any rule on ordering other then where multiple diff things (mixins, regular requires, other types) or concerned. My OCD only really cares that we get _one_  pattern and stick with it. I dunno how many are sorted which way currently.

claywar replied:
I tend to prefer alpha as well, though my personal approach is usually tree location, then alpha after.  I'm fine with any consistent approach so long as we only require things that are, well, required.

TeoTwawki replied:
tree then alpha sounds logical if we have anyone willing to do some mass editing later. for this pr for now: anythings acceptable we'll clean up later. I just took a quick glance and its gonna be a few hundred scripts needing cleanup for copyapasta

mrhappyasthma replied:
I'll double check that it's safe to remove 'settings.lua'.

Otherwise I'll leave the order the same for now (since it's consistent). We can write a script to sort the requires across the whole codebase.

claywar replied:
Just one note, would suggest not bulk sorting everything; there's a few instances where items are broken out on purpose (for example, quests and missions that use zone IDs)

TeoTwawki replied:
mobskill subdir should be safe to do though

mrhappyasthma committed:
Remove unused `require` of `settings`

mrhappyasthma commented:
Confirmed the `settings` require statement was not needed. Removed.

Xaver-DaRed commented:
One thing. From JP wiki, the blue spell is indeed a frontal attack, but the mob skill seems to be a rear attack.
It also seems to aply paralysis.

mrhappyasthma commented:
It applying paralysis does seem to be confirmed by all sources.

All sources also seem to confirm that it's cast anytime the target is behind. However some sources state that it can occur from the front as well.

E.g. "Often used when a player behind the mob pulls hate, but can still be used at any time." - https://ffxiclopedia.fandom.com/wiki/Sub-zero_Smash

Xaver-DaRed commented:
JP wiki seems to agree with that statement too.
https://55qyrhg5xpivgc2vfqirnmj7bu-adv7ofecxzh2qqi-wiki-ffo-jp.translate.goog/html/17588.html

It seems this mobs are not that smart.
